pkg/kubectl/cmd/run.go

```diff
@@ -196,7 +196,12 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 	}
 
 	params["env"] = cmdutil.GetFlagStringSlice(cmd, "env")
-
+	port := cmdutil.GetFlagInt(cmd, "port")
```

Comment:
`generateService` already has a check for the port value - can you add the extra checks there and get the same fix?